src/TuxArchitectStores.js

```diff
@@ -19,15 +19,15 @@ ArchitectChain.prototype.itNeeds = function (inputOrStore) {
   //if inputOrStore is array
   if (Array.isArray(inputOrStore)) {
     var inputOrStoreLength = inputOrStore.length;
-    //loop through inputOrStore array and push register id into waitForIds for each element in array
+    //concat each inputOrStore's __registerId__ into array
     for (var i = 0; i < inputOrStoreLength; i++) {
-      waitForIds.push(this._getRegisterId(inputOrStore[i]));
+      waitForIds = waitForIds.concat(this._getRegisterId(inputOrStore[i]));
     }
   } else {
-    //if inputOrStore is not an array push its corresponding register id into array
-    waitForIds.push(this._getRegisterId(inputOrStore));
+    //if inputOrStore is not an array then concat its corresponding register id into array
```

Comment:
Change comment to ...inputOrStore's \_\_registerId__ or set of \_\_registerId__'s...